quickwit-indexing/src/source/kinesis/shard_consumer.rs

```diff
@@ -0,0 +1,340 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+// TODO: Remove when `KinesisSource` is fully implemented.
+#![allow(dead_code)]
+
+use rusoto_kinesis::{Kinesis, Record};
+use tokio::sync::mpsc;
+use tokio::time::Duration;
+
+use crate::source::kinesis::api::{get_records, get_shard_iterator};
+
+#[derive(Debug)]
+enum ShardConsumerMessage {
+    NewShards(Vec<String>),
+    Records {
+        shard_id: String,
+        records: Vec<Record>,
+        lag_millis: Option<i64>,
+    },
+    /// The shard is closed after a merge or a split. There are no new records available.
+    ShardClosed(String),
+    /// The consumer has reached the latest record in the shard.
+    ShardEOF(String),
+}
+
+struct ShardConsumer {
+    stream_name: String,
+    shard_id: String,
+    starting_sequence_number: Option<String>,
+    eof_enabled: bool,
```

Comment:
A comment on the semantics of eof_enabled could be useful (the rest is fairly obvious)